File: kurae-web/src/lib/types/referral.ts
export type ReferralCode = {
  id: string;
  code: string;
  dropId?: string;
  dropTitle?: string;
  dropSlug?: string;
  clicksCount: number;
  signupsCount: number;
  ordersCount: number;
  createdAt: string;
};

export type ReferralStats = {
  valid: boolean;
  code?: string;
  clicksCount: number;
  signupsCount: number;
  ordersCount: number;
};

File: kurae-api/internal/domain/referral.go
package domain

type ReferralCode struct {
	ID           string  `json:"id"`
	Code         string  `json:"code"`
	DropID       *string `json:"dropId,omitempty"`
	DropTitle    *string `json:"dropTitle,omitempty"`
	DropSlug     *string `json:"dropSlug,omitempty"`
	ClicksCount  int     `json:"clicksCount"`
	SignupsCount int     `json:"signupsCount"`
	OrdersCount  int     `json:"ordersCount"`
	CreatedAt    string  `json:"createdAt"`
}
